### PR TITLE
Remove deprecated code scheduled for removal in x.25

### PR DESCRIPTION
**Pull request by nicktindall**
Hide whitespace for easier diff

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java [hunk 1]
```diff
@@ -650,78 +619,80 @@ public long exactExcerptsInCycle(int cycle) {
...
-        if (fromIndex > toIndex) {
-            long temp = fromIndex;
-            fromIndex = toIndex;
-            toIndex = temp;
-        }
-
-        // if the are the same
-        if (fromIndex == toIndex)
-            return 0;
+        try (ExcerptTailer tailer = createTailer()) {
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java:622]: This is less garbage-y than it was, but it's garbage-y. I wonder if we should provide an overload that takes a Tailer as a zero-garbage option, and have this one call it with a tailer it creates and closes.

Comment (rogersimmons) [reply src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java:622]: This feels like a reasonable use case for a thread-local which is lazy instantiated then reset and used when needed.

Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java:622]: Thread locals have caused all kinds of leakage problems and we have removed them elsewhere as customers ended up surprised. I think the overload is more intentional, even though it makes the API bigger

**Review by JerryShea — APPROVED** [review]
Approved. The countExcerpts question can be resolved later IMO

**Review by tgd — APPROVED** [review]

**Pull request merged**